CMakeLists.txt

```diff
@@ -129,9 +138,11 @@ target_link_libraries(monitor
 
 target_include_directories(monitor PRIVATE
   ${QWT_INCLUDE_DIR}
-  ${CMAKE_BINARY_DIR}
+  ${CMAKE_BINARY_DIR}/opendds-monitor
```

Comment:
This change is causing the build to fail on my end.

```
[ 58%] Building CXX object CMakeFiles/monitor.dir/src/graph_page.cpp.o
In file included from /home/fred/oci/opendds-monitor/src/graph_page.cpp:1:
/home/fred/oci/opendds-monitor/src/graph_page.h:52:10: fatal error: ui_graph_page.h: No such file or directory
   52 | #include "ui_graph_page.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```

It looks like it's for generated Qt ui header files. I'm guessing the location that needs to be included is different for you than for me and Tim since we both use Linux. Includes seem like the sort of thing that should be handled automatically, but apparently it's not for [`qt_wrap_ui`](https://cmake.org/cmake/help/latest/command/qt_wrap_ui.html). It says it's deprecated and to use [`AUTOUIC`](https://cmake.org/cmake/help/latest/prop_tgt/AUTOUIC.html) instead, which does handle the includes.

This works for me and I would assume for you as well:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index c75920a..41e3d43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,8 +95,6 @@ qt5_wrap_cpp(MOC_SOURCE
   src/topic_table_model.h
 )
 
-qt5_wrap_ui(UI_SOURCE ${UI})
-
 add_executable(monitor
   ${SOURCE}
   ${HEADER}
@@ -105,6 +103,9 @@ add_executable(monitor
   ${UI_SOURCE}
 )
 
+set_property(TARGET monitor PROPERTY AUTOUIC TRUE)
+set_property(TARGET monitor APPEND PROPERTY AUTOUIC_SEARCH_PATHS "${CMAKE_CURRENT_SOURCE_DIR}/ui")
+
 target_compile_features(monitor PRIVATE cxx_std_17)
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
@@ -138,7 +139,6 @@ target_link_libraries(monitor
 
 target_include_directories(monitor PRIVATE
   ${QWT_INCLUDE_DIR}
-  ${CMAKE_BINARY_DIR}/opendds-monitor
 )
 
 configure_file(opendds.ini . COPYONLY)
```

Reply:
Ah... that is more similar to how I have done it the past.  I will give it a try.